--- Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp.orig	2022-02-01 05:31:54.000000000 -0500
+++ Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp	2022-02-13 12:12:04.000000000 -0500
@@ -35,7 +35,7 @@
 #if ENABLE(JIT_CAGE)
 #include <WebKitAdditions/JITCageAdditions.h>
 #else // ENABLE(JIT_CAGE)
-#if OS(DARWIN)
+#if OS(DARWIN) && defined(MAP_JIT)
 #define MAP_EXECUTABLE_FOR_JIT MAP_JIT
 #define MAP_EXECUTABLE_FOR_JIT_WITH_JIT_CAGE MAP_JIT
 #else // OS(DARWIN)
@@ -45,7 +45,7 @@
 #endif // ENABLE(JIT_CAGE)
 
 #if OS(DARWIN)
-#include <wtf/spi/cocoa/MachVMSPI.h>
+#include <mach/vm_prot.h>
 #endif
 
 namespace WTF {
